go/vt/sqlparser/normalizer.go

```diff
@@ -185,13 +207,54 @@ func (nz *normalizer) convertLiteral(node *Literal, cursor *Cursor) {
 // and iterate on converting each individual value into separate
 // bind vars.
 func (nz *normalizer) convertComparison(node *ComparisonExpr) {
-	if node.Operator != InOp && node.Operator != NotInOp {
+	switch node.Operator {
+	case InOp, NotInOp:
+		nz.rewriteInComparisons(node)
+	default:
+		nz.rewriteOtherComparisons(node)
+	}
+}
+
+func (nz *normalizer) rewriteOtherComparisons(node *ComparisonExpr) {
+	col, ok := node.Left.(*ColName)
+	if !ok {
+		return
+	}
+	lit, ok := node.Right.(*Literal)
+	if !ok {
 		return
 	}
+	err := validateLiteral(lit)
+	if err != nil {
+		nz.err = err
+		return
+	}
+
+	bval := SQLToBindvar(lit)
+	if bval == nil {
+		return
+	}
+	key := keyFor(bval, lit)
+	bvname, ok := nz.vals[key]
+	if !ok || len(lit.Val) > 256 {
+		// If there's no such bindvar, or we have a big value, make a new one.
+		// Big values are most likely not for vindexes.
+		// We save a lot of CPU because we avoid building
+		bvname = nz.reserved.ReserveColName(col)
+		nz.vals[key] = bvname
+		nz.bindVars[bvname] = bval
+	}
```

Comment:
Shouldn't we return here?

Reply:
Nah, we do want to change the literal to a bindvar. I rewrote this method to be easier to read. let me know if I succeeded or not